.jenkins.groovy

```diff
@@ -0,0 +1,186 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+def deployableBranch = env.BRANCH_NAME ==~ /(1.7.x|1.8.x|main)/
+
+pipeline {
+
+    agent none
+
+    options {
+        // When we have test-fails e.g. we don't need to run the remaining steps
+        skipStagesAfterUnstable()
+        buildDiscarder(logRotator(numToKeepStr: '5', artifactNumToKeepStr: '5'))
+    }
+
+    stages {
+        stage('Build') {
+            matrix {
+                axes {
+                    axis {
+                        // https://cwiki.apache.org/confluence/display/INFRA/JDK+Installation+Matrix
+                        name 'MATRIX_JDK'
+                        values 'jdk_1.8_latest',
+                                'jdk_11_latest',
+                                'jdk_14_latest'
```

Comment:
Maybe add the other jdks as we have on main? Because OoenJ9 had given us headaches in the past.

Reply:
I don't disagree, but I'm trying to get the build stable first, I think there is a flakey test in here somewhere, but... it could have been something i tweaked too

Reply:
Latest JDK is 16 but I'm not sure we need to support non-LTS JDK

Reply:
Yes, this is just a build. It's not about support. We should test on as many JDKs as possible to find about problems with the next LTS as early as possible. So, -14, +16

Reply:
@bdemers the test I removed from main?

Reply:
It looked like a Groovy + ASM problem, so I didn't spend any more time with it

Reply:
@bdemers +1 to move forward ;)